### DISCUSSION TITLE
Smart Contract Lottery Subscription ID problem

### DISCUSSION THREAD

**Discussion by SankarNayak** [Cyfrin/foundry-full-course-cu#1734] [Q&A]
As Patrick Collins demonstrated in his video, to generate a random number, we need to go to the subscription manager in Chainlink. Here, we generate a subscription ID through a series of steps. We then deploy the Chainlink VRFv2 smart contract using the generated subscription ID. However, the VRFv2 contract requires a `uint64` subscription ID, whereas the generated ID is a `uint256` variable. On the other hand, VRFv2.5 accepts a `uint256` as a subscription ID. Given this discrepancy, what steps should I take to proceed?

**Comment by EngrPips** [comment]
Hello @SankarNayak, You can cast your subId to the type of uint64, as shown below. 

```js
uint256 subId = 100;
uint64 subIdToUse = uint64(subId);
```
So you can now use `subIdToUse` as the `subId` you will pass to `VRFv2`.

The other option is to keep watching the course until Patrick programmatically creates a sub. So, you make a subId programmatically instead of using the chainlink user interface. Doing it programmatically would natively return a type of `uint64` requested by the `VRFv2`. 

But what version are you working with? `VRFv2` or `VRFv2.5`?

**Reply by SankarNayak** [reply]
Thank you for your reply.

The issue is that Patrick used VRFv2 in his video, but at the time of his upload, the subID generated was of type uint64 which is used in VRFv2. Now, the subID generated is of type uint256 which is used for VRFv2.5. So my question is: should I use VRFv2.5 instead of VRFv2, or should I convert the subID as you suggested?
Also, will the conversion work?

**Reply by EngrPips** [reply]
Please use whatever you are okay with. And yeah, the conversion would work. I have seen it work for someone before.
